projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/notebook_step.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Callable
+
+from vdk.api.job_input import IJobInput
+from vdk.internal.builtin_plugins.run.step import Step
+
+log = logging.getLogger(__name__)
+
+# The function accept NotebookStep (below class) and IJobInput and
+# return true if the step has been executed and false if it is not (valid) executable step.
+# On error it's expected to raise an exception.
+NotebookStepFunction = Callable[["NotebookStep", IJobInput], bool]
+
+
+@dataclass
+class NotebookStep(Step):
```

Comment:
Why is the NotebookStep in a dedicated file, instead of moving next to `class Notebook` and calling it `notebook.py`? It seems as a compound part of `Notebook` concept, that consists of steps?